crates/fuel-gas-price-algorithm/src/v1/tests/algorithm_v1_tests.rs

```diff
@@ -143,3 +147,30 @@ fn worst_case__same_block_gives_the_same_value_as_calculate() {
     let expected = algorithm.calculate();
     assert_eq!(expected, actual);
 }
+
+#[test]
+fn da_change_should_not_panic() {
+    let updater = AlgorithmUpdaterV1 {
+        new_scaled_exec_price: 0,
+        min_exec_gas_price: 0,
+        exec_gas_price_change_percent: 0,
+        l2_block_height: 0,
+        l2_block_fullness_threshold_percent: ClampedPercentage::new(0),
+        new_scaled_da_gas_price: 0,
+        gas_price_factor: NonZeroU64::new(1).unwrap(),
+        min_da_gas_price: 0,
+        max_da_gas_price_change_percent: 0,
+        total_da_rewards_excess: 0,
+        da_recorded_block_height: 0,
+        latest_known_total_da_cost_excess: 0,
+        projected_total_da_cost: 0,
+        da_p_component: 0,
+        da_d_component: 0,
+        last_profit: 0,
+        second_to_last_profit: 0,
+        latest_da_cost_per_byte: 0,
+        unrecorded_blocks: vec![],
+    };
+
+    updater.da_change(i128::MIN / 2, i128::MIN / 2);
```

Comment:
Let's not test private functions. We should be able to test this with `update_l2_block_data`.

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2241/commits/3047c7e2024081212514b0bd3304ca37986fe732, however, I'm not fully convinced this fits better here.

_Rationale:_
we need to carefully select the values in order to trigger the "what used to be" an overflow inside `da_change()`. Now, if we, for example, change how the `total_da_rewards_excess` is updated [here](https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-gas-price-algorithm/src/v1.rs#L224-L227) or how we calculate P and D [here](https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-gas-price-algorithm/src/v1.rs#L311-L324) we may cause the `da_change()` to be invoked with values that do not cause overflow and the test will pass with both `abs()` and `saturating_abs()`.

I'm open to suggestions on how we can do this better.

Reply:
I'd be okay with removing the test since there are many places in the code that "could" panic if programmed poorly.

It might be worth adding some more aggressive prop testing for those cases.

Reply:
I think this boils down to the fact that people (me, at least) do not associate an "innocent" `abs()` call with a possible panic. Think `unwrap`s or `expect`s - these are usually caught during the review process and removed from the production code + after removing them, we do not add a test.

So yes, I agree that this test is kinda redundant - I removed it in https://github.com/FuelLabs/fuel-core/pull/2241/commits/7359bd7a850e6a48732c4bc887526dd39093f196

What we could do is:

1. Spawn a tech-debt issue to review the codebase against unwraps, expects, **abs** and similar and update relevant places
2. Rely on a clippy lint to detect those, but as far as I know there is no such lint. Maybe we can contribute to clippy and add "potential_panic_in_std"? :)

Wdyt?

Reply:
Currently we have
```rs
#![deny(clippy::arithmetic_side_effects)]
#![deny(clippy::cast_possible_truncation)]
#![deny(warnings)]
```

But I'd be open to adding more for sure.